File: src/main/java/no/nav/aap/api/søknad/pdl/PDLConfig.java
package no.nav.aap.api.søknad.pdl;

import no.nav.aap.api.søknad.rest.AbstractRestConfig;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.ConstructorBinding;
import org.springframework.boot.context.properties.bind.DefaultValue;

import java.net.URI;

@ConfigurationProperties(prefix = "pdl")
public class PDLConfig extends AbstractRestConfig {
    private static final String DEFAULT_BASE_URI = "http://pdl-api.pdl/graphql"; // må settes så lenge pdl er on prem
    private static final String DEFAULT_PING_PATH = "/";

    @ConstructorBinding
    public PDLConfig(@DefaultValue(DEFAULT_PING_PATH) String pingPath,
                     @DefaultValue("true") boolean enabled,
                     @DefaultValue(DEFAULT_BASE_URI) URI baseUri) {
        super(baseUri, pingPath, enabled);
    }
}

File: src/main/kotlin/no/nav/aap/api/oppslag/pdl/PDLConfig.kt
package no.nav.aap.api.oppslag.pdl

import java.net.URI
import java.time.Duration
import no.nav.aap.api.oppslag.pdl.PDLConfig.Companion.PDL
import no.nav.aap.rest.AbstractRestConfig
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.boot.context.properties.NestedConfigurationProperty
import org.springframework.boot.context.properties.bind.DefaultValue

@ConfigurationProperties(PDL)
@ConstructorBinding
class PDLConfig(baseUri: URI,
                @DefaultValue(DEFAULT_PING_PATH) pingPath: String,
                @NestedConfigurationProperty private val retryCfg: RetryConfig = RetryConfig.DEFAULT,
                @DefaultValue("true") enabled: Boolean) : AbstractRestConfig(baseUri, pingPath, PDL, enabled,retryCfg) {

    override fun toString() = "$javaClass.simpleName [baseUri=$baseUri, pingEndpoint=$pingEndpoint]"

    companion object {
        const val PDL_CREDENTIALS = "client-credentials-pdl"
        const val PDL = "pdl"
        const val DEFAULT_PING_PATH = ""
    }
}